src/App.jsx

```diff
@@ -14,45 +16,88 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function filterGoods(sortBy) {
+  const copy = [...goodsFromServer];
+
+  switch (sortBy) {
+    case SORT_BY_ALPHABET:
+      return copy.sort((good1, good2) => good1.localeCompare(good2));
+
+    case SORT_BY_LENGTH:
+      return copy.sort((good1, good2) => good1.length - good2.length);
+
+    default:
+      return copy;
+  }
+}
+
+const SORT_BY_ALPHABET = 'sortByAlphabet';
+const SORT_BY_LENGTH = 'sortByLength';
+
+export const App = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+
+  const preparedGoods = filterGoods(sortBy);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const resetSorting = () => {
+    setIsReversed(false);
+    setSortBy('');
+  };
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', ['is-info'], {
```

Comment:
It's still not fixed. Fix it everywhere, please.
![image](https://github.com/mate-academy/react_list-of-goods-js/assets/102293862/cde5ca4a-6adf-4637-859a-ab3ca69f86b4)